thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/auth/ThirdEyeAuthorizer.java

```diff
@@ -37,6 +38,14 @@ boolean authorize(
   // if the implementation does not support multiple namespace, please return a list with the value null Collections.singletonList(null)
   @NonNull List<String> listNamespaces(final ThirdEyePrincipal principal);
 
+  /**
+   * Generates the datasource connection configuration given principal
```

Comment:
maybe let's clarify the intent for our future selves. Especially given it's not the best design as discussed this morning.

```
Generates a recommended datasource configuration to use in the ThirdEye instance.
Does not create the datasource.

Example: an authorizer plugin could:
1. set up a service account to connect to Pinot 
2. return a valid datasource configuration that uses the service account just created.
```